libs/api/prisma/run-seed.ts

```diff
@@ -27,47 +28,52 @@ export async function runSeed() {
     throw new Error('@wepublish/api seeding has not been done')
   }
 
-  const admin = await prisma.user.findUnique({
-    where: {
-      id: 'admin'
+  const ensureAdminUserWithSetPassword = async (password: string) => {
+    const id = 'admin'
+    const userBase = {
+      email: 'admin@wepublish.ch',
+      name: 'Admin',
+      active: true,
+      roleIDs: [adminUserRole.id]
     }
-  })
-  if (!admin) {
-    const email = 'admin@wepublish.ch'
-    const overriddenPw = process.env.OVERRIDE_ADMIN_PW
-    let password = generateSecureRandomPassword(20)
-    if (overriddenPw) {
-      console.log(
-        '\x1b[31m\x1b[1m%s\x1b[0m',
-        `!!!!!!!!!!!!!!!!!! WARNING: UNSECURE PASSWORD IS OVERRIDDEN WITH ENV VARIABLE !!!!!!!!!!!!!!!!!! `
-      )
-      password = overriddenPw
-    }
-
-    console.log(
-      '\x1b[31m\x1b[1m%s\x1b[0m',
-      `Bootstrapping initial admin user ${email} with password: ${password}`
-    )
+    console.log('\x1b[31m\x1b[1m%s\x1b[0m', `Ensuring admin user`)
     await prisma.user.upsert({
-      where: {
-        id: 'admin'
+      where: {id},
+      update: {
+        ...userBase,
+        password: await bcrypt.hash(password, 11)
```

Comment:
Store password hash in a variable and use hash function just once.  And maybe increase the weight from 11 to 12

Reply:
OK